### DISCUSSION TITLE
Not able to upload CSV file to BiblioteQ.

### DISCUSSION THREAD

**Discussion by BiblKumar** [textbrowser/biblioteq#105] [Q&A] [answered]
When i am trying to import csv file to biblioteq, it is showing this particular error every time.
Database error (NOT NULL constraint failed: book.author Unable to fetch row) on row 1.
Database error (NOT NULL constraint failed: book.author Unable to fetch row) on row 2.
Sometimes it shows parmeter issue.
Didn't installed any software's related to it like Postgresql/sqlite/qt.
It is installed in Windows 7 32 bit.

Can you let me know how to solve that issue.

**Comment by textbrowser** [comment]
Are you attempting to import a file to BQ?

**Reply by BiblKumar** [reply]
![image](https://user-images.githubusercontent.com/89518086/130774015-86898616-fcde-4f1f-8d45-e162cce85a38.png)
yes

**Comment by textbrowser** [comment]
That particular error indicates that your CSV file or your configuration is missing the required database field author.

I will add a separate column to indicate required fields, although the errors already include that information. :)

**Comment by textbrowser** [comment]
There is a book template included in the import tool. I suggest using it as guidance. As BQ is open source, there is sufficient information regarding required (not null) database fields.

https://github.com/textbrowser/biblioteq/blob/master/Source/biblioteq_sqlite_create_schema.h

**Comment by textbrowser** [comment]
As there are many configurations, possibly infinitely-many, the implementation requires that you align your import file with the import tool. The tool can be improved upon, however, I'll need details regarding your specific needs.

For instance, I can add things such as:
- for missing authors, substitute a provided value
- ability to save and restore a specific configuration
- label required columns (mentioned above)

Let me know.

**Comment by textbrowser** [comment]
The tool was created because it was requested many years ago. The particular person who requested it vanished and did not provide an evaluation, guidance, or preferences. So, the current form was my interpretation of their needs. :)

**Reply by BiblKumar** [reply]
Tool is still good, I like that. I run alone a small library here with 8k books..so it is not easy to add each and every book manually, so, I am just trying import csv file with two columns(Title & Accession Number) at that time facing issues. Now it is showing parameter count mismatch.
I just want to upload the two columns(title & accession number) that is more than enough for me. Please let me know how to do it

**Comment by textbrowser** [comment]
Is your CSV file private? If not, paste it here and I'll create the database. Will upload a video of my results.
You can also attach it.

**Reply by BiblKumar** [reply]
PFA.
[s.csv](https://github.com/textbrowser/biblioteq/files/7046546/s.csv)

**Comment by textbrowser** [comment]
OK. It only includes accession numbers and titles. I'll have to modify BQ. ISBNs are not included so they'll be ignored. Right, so there is some work required. May take some time. Stay tuned?

**Comment by textbrowser** [comment]
[s.csv](https://github.com/textbrowser/biblioteq/files/7051134/s.csv)
[test.sqlite.zip](https://github.com/textbrowser/biblioteq/files/7051145/test.sqlite.zip)

Here you are. Please let me know the results.

I also rearranged some of the columns in your CSV file.

**Comment by BiblKumar** [comment]
PFA screen shots
![image](https://user-images.githubusercontent.com/89518086/130906108-49b02ecf-f6c8-446f-97f3-7b432331263c.png)
![image](https://user-images.githubusercontent.com/89518086/130906226-e2351071-6063-40b8-a592-fb4c2185476e.png)
it is the same. I think need all the data for every column?

**Comment by textbrowser** [comment]
The new changes have not been released. Of course, every column is preferred. Otherwise, many nonsensical values.

**Reply by BiblKumar** [reply]
So, I filled up every row and column with 'N/A' where there is no data. Still i got this error, 
![image](https://user-images.githubusercontent.com/89518086/130954025-b50a1ec6-793e-4600-b9a7-2ac5797cdb45.png)

**Comment by textbrowser** [comment]
This error indicates duplicate ISBN-13 values.

**Reply by BiblKumar** [reply]
It seems I have to enter each and every book manually....Thanks for the reply.

**Comment by textbrowser** [comment, marked as answer]
Yup. The import is just that. However, if you have the ISBNs, I can change the import to create the ISBN-13s automatically. Again, it'll be a software change. If you include your current CSV, I can do that.

**Reply by BiblKumar** [reply]
Your Tool is good...it is taking time for me to upload the each n every book, but, it's okay to me.
Please can you add one more option in the members data, it is allowing to take any number of books or magazines but not restricting the member to take these many only. If you can add that option in the members database  from restricting the member to take 1 to 10 or 20, it would be good to us, so, every time we don't need to check it. Please, can you let me know is that possible to make the changes in the software tool.
